### PR TITLE
feat: parse big-number string as BigInt

### PR DESCRIPTION
**Pull request by juanluispaz**
This pull request fix the issue #68

### CODE REVIEW

**Review by niieani — CHANGES_REQUESTED** [review]
May I ask you to rebase, use prettier formatting and remove the built files from the commit?
Thanks.

**Comment by simontaisne** [comment]
Any chance this PR lands?

**Comment by niieani** [comment]
Thanks @simontaisne, feel free to address my comments to @juanluispaz's PR and submit another PR, I'll be happy to merge and release it.

File: lib/hashids.ts [hunk 1]
```diff
@@ -374,15 +374,31 @@ const fromAlphabet = (
...
+
+  const n = parseInt(str, 10)
+  if (Number.isSafeInteger(n)) {
+    return n
+  } else if (typeof BigInt === 'function') {
+    return BigInt(str)
+  } else {
+    // we do not have support for BigInt:
+    throw new Error(
+      `Unable to encode the provided big-number string without lose information, due to lack of support for BigInt numbers in the current environment`,
```
Comment (niieani) [inline lib/hashids.ts:390]:
```suggestion
      `Unable to encode the provided BigInt string without loss of information due to lack of support for BigInt type in the current environment`,
```

**Comment by niieani** [comment]
superseded by https://github.com/niieani/hashids.js/pull/546

**Pull request closed without merging**